### PR TITLE
return a single value for seq when to == from, fixes #25

### LINKED ISSUE #25: seq.integer64() doesn't match base seq() when from and to are identical
`seq.integer64()` doesn't match `seq()` when from and to are identical:

```r
> seq(1, 1)
[1] 1
> library(bit64)
> seq(as.integer64(1), as.integer64(1))
integer64(0)                                                                                                                                                                                                  
Warning message:                                                                                                                                                                                              
In `%/%.integer64`((to - from), by) : NAs produced due to division by zero 
```

sessionInfo():

```r
R version 4.2.1 (2022-06-23)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Debian GNU/Linux 11 (bullseye)

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_DK.UTF-8        LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8      
 [8] LC_NAME=C                  LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] bit64_4.0.5 bit_4.0.4  

loaded via a namespace (and not attached):
[1] compiler_4.2.1
```

### PR DESCRIPTION
**Pull request by cswingle**
Fixes issue #25 where using the same from and to result in a division by zero error and an incorrect result (at least compared with the base integer `seq()` function).

### CODE REVIEW

**Comment by MichaelChirico** [comment]
Thanks @cswingle & sorry for the delay! Change basically LGTM. Would you mind writing a NEWS entry?

File: R/integer64.R [hunk 1]
```diff
@@ -2370,7 +2370,11 @@ seq.integer64 <- function(from=NULL, to=NULL, by=NULL, length.out=NULL, along.wi
     }
 
     if (!length.out){
-      length.out <- (to-from) %/% by + 1L
+      if (to == from) {
```
Comment (MichaelChirico) [inline R/integer64.R:2373]: IINM the crux here is that `by==0`, not that `to==from`, it's just that we've inferred `by==0` due to `to==from`.

Might it not be clearer for that to be the condition, then?
```suggestion
      if (by != 0L) {
```

**Comment by MichaelChirico** [comment]
BTW I also found #47 in case you're looking for a similar place to contribute :wink:

**Comment by cswingle** [comment]
> Thanks @cswingle & sorry for the delay! Change basically LGTM. Would you mind writing a NEWS entry?

It looks like you did this already in NEWS.0?

**Comment by MichaelChirico** [comment]
Oh, i wasn't me there but I see what you mean. Hmm, I also see your issue appears fixed on `master`:

```r
seq(as.integer64(1), as.integer64(1))
# integer64
# [1] 1
```

So I take it this PR was addressed in a slightly different way based on the R/integer64.R changes I'm seeing in d60b6c5b45fed915a607aaca3784e0dde1b1dfbe. I also only just noticed the linked issue is already closed :)

There should be a regression test, but that'll come at some point with the rapidly growing suite.

Thank you for your contribution!

**Pull request closed without merging**